modules/distribution/src/main/lib/fetch.sh

```diff
@@ -0,0 +1,108 @@
+#! /usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+# agreements. See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the License. You may obtain a
+# copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+lib_dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+maven_prefix=https://repo1.maven.org/maven2
+
+function download {
+  IFS=':' read -ra DEP <<< "$1" 
+  dir=$lib_dir/
+  if [ -n "$2" ]; then
+    dir=$lib_dir/$2
+    if [ ! -d $dir ]; then
+      mkdir $dir
+    fi
+  fi
+  group=${DEP[0]}
+  artifact=${DEP[1]}
+  ftype=${DEP[2]}
+  version=${DEP[3]}
+  fn=$artifact-$version.$ftype
+  path="${group//.//}/$artifact/$version/$fn"
+  download_url=$maven_prefix/$path
+
+  if [ -f $dir/$fn ]; then
+    echo "SUCCESS: Dependency exists - $dir/$fn"
+  else 
+    wget -q $download_url -P $dir
+    if [ $? == 0 ]; then
+      echo "SUCCESS: Dependency downloaded from $download_url"
+    else
+      echo "ERROR: Dependency download failed - $download_url"
+    fi
+  fi
+}
+
+case "$1" in
+ahz)
+  echo "Fetching Hadoop, Accumulo, & Zookeeper dependencies"
+  cd $lib_dir/ahz
+  rm -f *.jar
+  mvn clean package ${*:2}
+  ;;
+extra)
+  echo "Fetching extra Fluo dependencies"
```

Comment:
Sounds good to me.  I put it after the download of the dependencies so users will see it.  See 6c211b3e4ff54